coderd/workspaceapps.go

```diff
@@ -95,6 +95,7 @@ func (api *API) appHost(rw http.ResponseWriter, r *http.Request) {
 func (api *API) workspaceAppsProxyPath(rw http.ResponseWriter, r *http.Request) {
 	workspace := httpmw.WorkspaceParam(r)
 	agent := httpmw.WorkspaceAgentParam(r)
+	apiKey := httpmw.APIKey(r)
```

Comment:
I'm getting the API Key but I'm concerned with the use case where this app is public and the current user is not authenticated. Is there a way to verify if the apiKey in this case is null?

Reply:
Further down in this handler we have `_, hasAPIKey := httpmw.APIKeyOptional(r)` which you could just move up to the top of the function. Then just wrap the stats code in `if hasAPIKey { // upload stats }`